### PR TITLE
Code was refactored globally

### PR DESCRIPTION
**Pull request by mpushnikA1QA**

### CODE REVIEW

File: src/test/features/checkNextPageAndPreviousPageFields.feature [hunk 1]
```diff
@@ -23,7 +23,7 @@ Feature: NextPage and PreviousPage fields in response body for search and feed e
             | organizationHierarchy | feed        |
 
     Scenario Outline: Send a request to [<entity>] entity to [<apiEndpoint>] endpoin and check that nextPage field is present and work
-        When Send a request to '<entity>' entity common search endpoint with random search parameter for next page field
+        When Send a request to the '<entity>' entity common search endpoint with random search parameter for next page field
         Then The API status code should be 200 - OK
         And There is a only nextPage field in the API response for '<apiEndpoint>' endpoint for the '<entity>' entity and response has the correct json schema
         And Move to the nextPage link for the '<apiEndpoint>' endpoint
```
Comment (danil-shamsiev) [inline src/test/features/checkNextPageAndPreviousPageFields.feature:29]: Change `And` to `When`

File: src/test/features/checkNextPageAndPreviousPageFields.feature [hunk 2]
```diff
@@ -39,7 +39,7 @@ Feature: NextPage and PreviousPage fields in response body for search and feed e
             | organization | search      |
 
     Scenario Outline: Send a request to [<entity>] entity to [<apiEndpoint>] endpoin and check that previous field is present and work
-        When Send a request to '<entity>' entity common search endpoint with random search parameter for next page field
+        When Send a request to the '<entity>' entity common search endpoint with random search parameter for next page field
         Then The API status code should be 200 - OK
         And Move to the nextPage link for the '<apiEndpoint>' endpoint
```
Comment (danil-shamsiev) [inline src/test/features/checkNextPageAndPreviousPageFields.feature:44]: Change `And` to `When`

File: src/test/features/checkNextPageAndPreviousPageFields.feature [hunk 3]
```diff
@@ -39,7 +39,7 @@ Feature: NextPage and PreviousPage fields in response body for search and feed e
...
         And The API status code should be 200 - OK
```
Comment (danil-shamsiev) [inline src/test/features/checkNextPageAndPreviousPageFields.feature:45]: Change `And` to `Then`

File: src/test/steps/checkNextPageAndPreviousPageFields.js [hunk 4]
```diff
@@ -18,54 +18,43 @@ const FEED_VALUE = "feed";
...
+  this.sharedHost = response.request.host;
+  this.sharedResponseBody = response.data;
 });
 
-Then(/^Move to the nextPage link for the '(.*)' endpoint$/, async (apiEndpoint) => {
-  const actualApiEndpoint = getApiEndpoint(apiEndpoint);
-  const actualResponseData = actualApiEndpoint === FEED_VALUE ? $feedResponseBody : this.searchResponseBody;
-  const nextPageCommonUrl = actualResponseData.pagination.nextPage;
-  const nextPageUrl = nextPageCommonUrl.replace($host, "").replace(HTTP_PART, "");
+Then(/^Move to the nextPage link for the '(.*)' endpoint$/, async function (apiEndpoint) {
```
Comment (danil-shamsiev) [inline src/test/steps/checkNextPageAndPreviousPageFields.js:32]: Change `Then` to `When` as there is no assertion

File: src/test/steps/checkNextPageAndPreviousPageFields.js [hunk 5]
```diff
@@ -18,54 +18,43 @@ const FEED_VALUE = "feed";
...
+  const nextPageCommonUrl = this.sharedResponseBody.pagination.nextPage;
+  const nextPageUrl = nextPageCommonUrl.replace(this.sharedHost, "").replace(HTTP_PART, "");
   const response = await $feedEndpointNextPreviousPageAPIClient.getRequest(nextPageUrl);
-  $statusCode = response.status;
+  this.sharedStatusCode = response.status;
   this.nextPageResponseData = response.data;
 });
 
-Then(/^Move to the previousPage link$/, async () => {
+Then(/^Move to the previousPage link$/, async function () {
```
Comment (danil-shamsiev) [inline src/test/steps/checkNextPageAndPreviousPageFields.js:40]: Change `Then` to `When` as there is no assertion

File: src/test/steps/commonSteps.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    const keysObjectArray = Object.keys(item);
+    const diff = getCommonValuesBetweenArrays(keysObjectArray, this.sharedExpectedItemsParameters);
+    if (diff.length > 0) {
+      nonMatchingFieldsArray.push(`There shouldn't be fields [${diff}] for ${entityId} - [${item[entityId]}]`);
+    }
+  }
+  expectChai(nonMatchingFieldsArray.length).to.equal(0, `The fields specified in the request query shouldn't be in the response body : ${nonMatchingFieldsArray.toString()}`);
+});
+
+//TODO - It is necessary to do code refactoring for the entire project.
```
Comment (danil-shamsiev) [inline src/test/steps/commonSteps.js:116]: Is this comment relevant anymore?

**Review by danil-shamsiev — CHANGES_REQUESTED** [review]

File: src/test/features/checkNextPageAndPreviousPageFields.feature (hunk 1 above)
Comment (mpushnikA1QA) [reply src/test/features/checkNextPageAndPreviousPageFields.feature]: Fixed

File: src/test/features/checkNextPageAndPreviousPageFields.feature (hunk 2 above)
Comment (mpushnikA1QA) [reply src/test/features/checkNextPageAndPreviousPageFields.feature]: Done

File: src/test/features/checkNextPageAndPreviousPageFields.feature (hunk 3 above)
Comment (mpushnikA1QA) [reply src/test/features/checkNextPageAndPreviousPageFields.feature]: Fixed

File: src/test/steps/checkNextPageAndPreviousPageFields.js (hunk 4 above)
Comment (mpushnikA1QA) [reply src/test/steps/checkNextPageAndPreviousPageFields.js]: updated

File: src/test/steps/checkNextPageAndPreviousPageFields.js (hunk 5 above)
Comment (mpushnikA1QA) [reply src/test/steps/checkNextPageAndPreviousPageFields.js]: Fixed

File: src/constants/apiEndpoints.js [hunk 7]
```diff
@@ -1,13 +1,13 @@
-//TODO: The best way to add v1 for the parameters
-const START_VERSION = 'v1';
-const FEED_ENTITY_ENDPOINT = (entity) => `/${START_VERSION}/feed/${entity}`;
-const SEARCH_ENTITY_ENDPOINT = (entity) => `/${START_VERSION}/search/${entity}`;
+const { apiVersion } = require("../environment/api");
+
+const FEED_ENTITY_ENDPOINT = (entity) => `/${apiVersion}/feed/${entity}`;
+const SEARCH_ENTITY_ENDPOINT = (entity) => `/${apiVersion}/search/${entity}`;
```
Comment (danil-shamsiev) [inline src/constants/apiEndpoints.js:4]: Format `apiVersion` as UPPER_SNAKE_CASE

**Review by danil-shamsiev — CHANGES_REQUESTED** [review]

**Review by danil-shamsiev — APPROVED** [review]

**Pull request merged**